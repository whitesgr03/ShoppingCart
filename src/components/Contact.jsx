import "../style/contact.css";
import Icon from "@mdi/react";
import { mdiPhone, mdiEmailOutline, mdiMapMarker } from "@mdi/js";

import { useOutletContext } from "react-router-dom";

const Contact = () => {
	const { imageUrls } = useOutletContext();

	const url = imageUrls.find(imageUrl => imageUrl.includes("contact"));

	return (
		<div className="contact">
			<div
				style={{
					backgroundImage: url ? `url(${url})` : " none",
				}}
			>
				<div className="blur">
					<h2 className="title">Contact Us</h2>
					<fieldset className="container">
						<legend>Get in touch</legend>
						<div className="info">
							<ul className="icons">
								<li>
									<Icon path={mdiPhone} size={1} />
								</li>
								<li>
									<Icon path={mdiEmailOutline} size={1} />
								</li>
								<li>
									<Icon path={mdiMapMarker} size={1} />
								</li>
							</ul>
							<ul className="describe">
								<li>+123-456-789</li>
								<li>gentskin@example.com</li>
								<li>123 Anywhere St.,Any City</li>
							</ul>
						</div>
						<form id="contact" onSubmit={e => e.preventDefault()}>
							<label htmlFor="name">
								Name
								<input type="text" name="name" id="name" />
							</label>
							<label htmlFor="email">
								Email
								<input type="text" name="email" id="email" />
							</label>
							<label htmlFor="content">
								Content
								<textarea
									name="content"
									id="content"
									rows="5"
								></textarea>
							</label>
						</form>
					</fieldset>
					<div className="wrap">
						<button type="submit" form="contact" className="slide">
							Send Message
						</button>
					</div>
				</div>
			</div>
		</div>
	);
};

export { Contact as default };
