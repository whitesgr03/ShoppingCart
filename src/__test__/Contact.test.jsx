import { render } from "@testing-library/react";

import Contact from "../components/Contact";

describe("Renders Contact Component", () => {
	it("Should return Contact DOM", () => {
		const { container } = render(<Contact />);

		expect(container).toMatchSnapshot();
	});
});
