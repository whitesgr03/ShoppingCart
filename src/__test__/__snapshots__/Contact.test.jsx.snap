// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Contact Component Should return Contact DOM 1`] = `
<div>
  <div
    class="contact"
  >
    <div>
      <h2
        class="title"
      >
        Contact Us
      </h2>
      <fieldset
        class="container"
      >
        <legend>
          Get in touch
        </legend>
        <div
          class="info"
        >
          <ul
            class="icons"
          >
            <li>
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.62,10.79C8.06,13.62 10.38,15.94 13.21,17.38L15.41,15.18C15.69,14.9 16.08,14.82 16.43,14.93C17.55,15.3 18.75,15.5 20,15.5A1,1 0 0,1 21,16.5V20A1,1 0 0,1 20,21A17,17 0 0,1 3,4A1,1 0 0,1 4,3H7.5A1,1 0 0,1 8.5,4C8.5,5.25 8.7,6.45 9.07,7.57C9.18,7.92 9.1,8.31 8.82,8.59L6.62,10.79Z"
                  style="fill: currentColor;"
                />
              </svg>
            </li>
            <li>
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 6C22 4.9 21.1 4 20 4H4C2.9 4 2 4.9 2 6V18C2 19.1 2.9 20 4 20H20C21.1 20 22 19.1 22 18V6M20 6L12 11L4 6H20M20 18H4V8L12 13L20 8V18Z"
                  style="fill: currentColor;"
                />
              </svg>
            </li>
            <li>
              <svg
                role="presentation"
                style="width: 1.5rem; height: 1.5rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,11.5A2.5,2.5 0 0,1 9.5,9A2.5,2.5 0 0,1 12,6.5A2.5,2.5 0 0,1 14.5,9A2.5,2.5 0 0,1 12,11.5M12,2A7,7 0 0,0 5,9C5,14.25 12,22 12,22C12,22 19,14.25 19,9A7,7 0 0,0 12,2Z"
                  style="fill: currentColor;"
                />
              </svg>
            </li>
          </ul>
          <ul
            class="describe"
          >
            <li>
              +123-456-789
            </li>
            <li>
              gentskin@example.com
            </li>
            <li>
              123 Anywhere St.,Any City
            </li>
          </ul>
        </div>
        <form
          id="contact"
        >
          <label
            for="name"
          >
            Name
            <input
              id="name"
              name="name"
              type="text"
            />
          </label>
          <label
            for="email"
          >
            Email
            <input
              id="email"
              name="email"
              type="text"
            />
          </label>
          <label
            for="content"
          >
            Content
            <textarea
              id="content"
              name="content"
              rows="5"
            />
          </label>
        </form>
      </fieldset>
      <div
        class="wrap"
      >
        <button
          class="slide"
          form="contact"
          type="submit"
        >
          Send Message
        </button>
      </div>
    </div>
  </div>
</div>
`;
